source/common/router/rds_impl.cc

```diff
@@ -174,5 +183,42 @@ Router::RouteConfigProviderSharedPtr RouteConfigProviderManagerImpl::getRouteCon
   return new_provider;
 };
 
+void RouteConfigProviderManagerImpl::addRouteInfo(RdsRouteConfigProviderSharedPtr provider,
+                                                  Buffer::Instance& response) {
+  response.add(fmt::format("route_config_name: {}\n", provider->routeConfigName()));
+  response.add(fmt::format("cluster_name: {}\n", provider->clusterName()));
+  response.add("config dump:\n");
+  response.add(fmt::format("{}\n", provider->configAsString()));
```

Comment:
+1 to just making the output JSON from the start. It should be relatively easy to do. I think the proto can be converted to JSON also @htuch ?